### PR TITLE
Problem: Missing primitives for implementing remote attestation

### PR DESCRIPTION
**Pull request by devashishdxt**
Solution: Added crypto primitives in `enclave-utils`. Related to #817.

### CODE REVIEW

**Comment by devashishdxt** [comment]
> tests?

I'll add them in future PRs. I'm breaking down one big PR in three PRs.
- Add primitives
- Implement `ra-sp`
- Add tests

File: chain-tx-enclave/enclave-utils/src/crypto/key_exchange.rs [hunk 1]
```diff
@@ -0,0 +1,123 @@
+use std::io::Write;
```
Comment (tomtau) [inline chain-tx-enclave/enclave-utils/src/crypto/key_exchange.rs]: when is this custom key exchange used?

**Comment by tomtau** [comment]
without tests or comments, it's hard to tell whether this solves the problem it claims to solve

File: chain-tx-enclave/enclave-utils/src/crypto/key_exchange.rs (hunk 1 above)
Comment (devashishdxt) [reply chain-tx-enclave/enclave-utils/src/crypto/key_exchange.rs]: https://github.com/ndokmai/rust-sgx-remote-attestation/blob/7e2c26930d4a87aa040b3e1d5602c61fcd5145ee/ra-sp/src/context.rs#L142

**Comment by devashishdxt** [comment]
> without tests or comments, it's hard to tell whether this solves the problem it claims to solve

Ok. I'll add tests in this PR.

File: chain-tx-enclave/enclave-utils/src/crypto/key_exchange.rs (hunk 1 above)
Comment (tomtau) [reply chain-tx-enclave/enclave-utils/src/crypto/key_exchange.rs]: it's copying that example with a modified Sigma protocol: https://github.com/intel/sgx-ra-sample I guess?
Which component will need a communication over Sigma? Here, the end goal is to have an attested TLS client<->server channel, so one needs HTTPS to IAS, processing its payloads, AESM comms + verification of quotes (checking CMAC, SHA256(nonce||quote)... )

**Comment by tomtau** [comment]
closing for now, as @devashishdxt will submit a new PR later

**Pull request closed without merging**